mobile/src/context/CustomerContext.tsx

```diff
@@ -90,37 +91,42 @@ export function CustomerProvider({
   }
 
   async function updateCustomerState() {
-    try {
-      const customerResponse = await getCustomer();
-
-      if (customerResponse.data.value.status === "UNDERREVIEW") {
+    const customerResponse = await getCustomer();
+    if (!isNil(customerResponse)) {
+      if (customerResponse?.data.value.status === "UNDERREVIEW") {
         dispatch({
           type: CustomerStateActionType.UPDATE_STATE,
           state: CustomerStateType.CUSTOMER_UNDER_REVIEW,
+          is_business: customerResponse?.data.value.isBusiness,
         });
         return false;
       }
-      return true;
-    } catch (error) {
-      const axiosError = error as AxiosError<APIError>;
-
-      if (axiosError.response?.status === 404 ||
-        axiosError.response?.status === 401) {
+      if (customerResponse?.data.value.status === "BLOCKED") {
         dispatch({
           type: CustomerStateActionType.UPDATE_STATE,
-          state: CustomerStateType.CUSTOMER_REQUIRED,
+          state: CustomerStateType.CUSTOMER_UNDER_REVIEW,
+          is_business: customerResponse?.data.value.isBusiness,
         });
         return false;
       }
-
-      dispatch({
-        type: CustomerStateActionType.ERROR,
-        errorMessage:
-          error instanceof AxiosError && axiosError.response?.data.Errors[0]
-            ? axiosError.response.data.Errors[0].Message
-            : (error as Error).message,
-      });
-
+      if (customerResponse?.data.value.status === "NEW") {
```

Comment:
Can we rather use or create new Enums for these states? We are already using enums in other compares in this class for example `isNew: state.state ? state.state === CustomerStateType.CUSTOMER_NEW : null,`

Reply:
Fixed.